src/App.tsx

```diff
@@ -8,55 +8,148 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/users';
+import { User } from './types/User';
+import { getPosts } from './api/posts';
+import { Post } from './types/Post';
+import { getComments } from './api/commets';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [userSelect, setUserSelect] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const [isPostError, setIsPostError] = useState<boolean>(false);
+  const [isPostsLoading, setIsPostsLoading] = useState<boolean>(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+
+  const [postIdComments, setPostIdComments] = useState<number | null>(null);
+  const [isCommentsLoading, setIsCommentsLoading] = useState<boolean>(false);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [isCommentError, setIsCommentError] = useState<boolean>(false);
+
+  const [isNewComment, setIsNewComment] = useState<boolean>(false);
+
+  const fetchUsers = useCallback(async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUserSelect(usersFromServer);
+    } catch (error) {
+      setIsPostError(true);
+    }
+  }, [userSelect]);
```

Comment:
As you don't include this function in any deps array (but you should. See other comments) and you don't pass it to children you don't need useCallback. In this case it's even bad for performance. I highly recommend to read this article to understand how useCallback works.
https://beta.reactjs.org/apis/react/useCallback